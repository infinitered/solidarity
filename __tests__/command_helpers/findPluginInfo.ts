const findPluginInfo = require('../../src/extensions/functions/findPluginInfo')
const examplePlugin = require('examplePlugin')
const mockContext = examplePlugin(require('mockContext'))

describe('findPluginInfo Function', () => {
  test('can fail to find plugins', () => {
    const rule = {
      rule: 'custom',
      plugin: 'FAKE',
      name: 'checkThing',
    }
    const customPluginRule = findPluginInfo(rule, mockContext)
    expect(customPluginRule).toEqual({ message: "Plugin not found 'FAKE'", success: false })
  })

  test('can fail to find plugins', () => {
    const rule = {
      rule: 'custom',
      plugin: 'Example Plugin',
      name: 'FAKE',
    }
    const customPluginRule = findPluginInfo(rule, mockContext)
    expect(customPluginRule.message).toEqual(
      "NOT FOUND: Custom rule from 'Example Plugin' plugin with check function 'FAKE'"
    )
    expect(customPluginRule.success).toBeFalsy()
  })

  test('can find plugins', () => {
    const rule = {
      rule: 'custom',
      plugin: 'Example Plugin',
      name: 'checkThing',
    }
    const customPluginRule = findPluginInfo(rule, mockContext)
    expect(customPluginRule.success).toBeTruthy()
    expect(customPluginRule).toMatchSnapshot()
  })
})
