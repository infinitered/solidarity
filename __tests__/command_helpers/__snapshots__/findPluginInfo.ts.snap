// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`findPluginInfo Function can find plugins 1`] = `
Object {
  "plugin": Object {
    "check": [Function],
    "report": [Function],
    "snapshot": [Function],
  },
  "success": true,
}
`;
